src/Microsoft.Extensions.Configuration.AzureAppConfiguration/AzureAppConfigurationSource.cs

```diff
@@ -11,15 +11,19 @@ internal class AzureAppConfigurationSource : IConfigurationSource
         private readonly bool _optional;
         private readonly Func<AzureAppConfigurationOptions> _optionsProvider;
         private readonly IConfigurationClientFactory _configurationClientFactory;
+        private readonly string _environmentName;
+        private readonly bool _environmentNameEnabled;
 
-        public AzureAppConfigurationSource(Action<AzureAppConfigurationOptions> optionsInitializer, bool optional = false, IConfigurationClientFactory configurationClientFactory = null)
+        public AzureAppConfigurationSource(Action<AzureAppConfigurationOptions> optionsInitializer, string environmentName = null, bool environmentNameEnabled = false, bool optional = false, IConfigurationClientFactory configurationClientFactory = null)
```

Comment:
We dont need `environmentNameEnabled` flag. We can check `environmentName == null` wherever needed.

Reply:
I guess I should've asked before, but is passing null for environmentName not a valid option? In other words, is there no scenario where the environment variables would both be null and the user could pass that in to check specifically for null? That was the only reason I added this.

Reply:
From usage perspective, "null" environment doesn't make sense. Dotnet also defaults to "Production" environment if the environment is not explicitly set ([reference](https://learn.microsoft.com/en-us/aspnet/core/fundamentals/environments?view=aspnetcore-7.0#environments)). 
So I think we should explicitly forbid null as an environment name.